src/script/lua_api/l_mapgen.cpp

```diff
@@ -1773,6 +1773,31 @@ int ModApiMapgen::l_place_schematic_on_vmanip(lua_State *L)
 	return 1;
 }
 
+// spawn_tree_on_vmanip(vmanip, pos, treedef)
+int ModApiMapgen::l_spawn_tree_on_vmanip(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	MMVManip *vm = checkObject<LuaVoxelManip>(L, 1)->vm;
+	v3s16 p0 = read_v3s16(L, 2);
+	treegen::TreeDef tree_def;
+	const NodeDefManager *ndef = getGameDef(L)->ndef();
+	if (!read_tree_def(L, 3, ndef, tree_def))
+		return 0;
+
+	treegen::error e = treegen::make_ltree(*vm, p0, tree_def);
```

Comment:
what happens if p0 is out of bounds?

Reply:
The tree still gets calculated outside the vmanip, but only parts of the tree which spread into the vmanip will be placed.